plugins/woocommerce-admin/client/payments-welcome/apms.tsx

```diff
@@ -0,0 +1,137 @@
+/**
+ * External dependencies
+ */
+import { List } from '@woocommerce/components';
+import { download, Icon } from '@wordpress/icons';
+
+import {
+	Card,
+	CardBody,
+	CardFooter,
+	CardHeader,
+	ExternalLink,
+	Notice,
+	ToggleControl,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import strings from './strings';
+import { WC_ASSET_URL } from '~/utils/admin-settings';
+
+export interface Apm {
+	id: string;
+	title: string;
+	icon: string;
+	description: string;
+	link: string;
+	extension: string;
+}
+
+export const apms: Apm[] = [
+	{
+		id: 'paypal',
+		title: strings.apms.paypal.title,
+		icon: `${ WC_ASSET_URL }images/payment_methods/72x72/paypal.png`,
+		description: strings.apms.paypal.description,
+		link: 'https://woocommerce.com/products/woocommerce-paypal-payments/',
+		extension: 'woocommerce-paypal-payments',
+	},
+	{
+		id: 'amazonpay',
+		title: strings.apms.amazonpay.title,
+		icon: `${ WC_ASSET_URL }images/payment_methods/72x72/amazonpay.png`,
+		description: strings.apms.amazonpay.description,
+		link: 'https://woocommerce.com/products/pay-with-amazon/',
+		extension: 'woocommerce-gateway-amazon-payments-advanced',
+	},
+	{
+		id: 'klarna',
+		title: strings.apms.klarna.title,
+		icon: `${ WC_ASSET_URL }images/payment_methods/72x72/klarna.png`,
+		description: strings.apms.klarna.description,
+		link: 'https://woocommerce.com/products/klarna-payments/',
+		extension: 'klarna-payments-for-woocommerce',
+	},
+];
+
+interface ApmsProps {
+	enabledApms: Set< Apm >;
+	setEnabledApms: ( value: Set< Apm > ) => void;
+}
+
+const ApmNotice = ( { enabledApms }: { enabledApms: Set< Apm > } ) => {
+	if ( ! enabledApms.size ) {
+		return null;
+	}
+
+	const extensions = [ ...enabledApms ]
```

Comment:
Hey @oaratovskyi Is there any specific reason that we need to use `[ ... ]` instead of `enabledApms.map` directly? I saw you use `[ ... ].map` in many places.

Reply:
Hi @chihsuan!
As we’re storing it in a `Set` to be able to use `map` or `filter` we first need to convert the set to an array.

Reply:
Ah, okay, make sense to me!